File: 1984-maximum-distance-between-a-pair-of-values/maximum-distance-between-a-pair-of-values.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var maxDistance = function (nums1, nums2) {
    let distance = 0;
    let i = 0, j = 0;

    while (j < nums2.length && i < nums1.length) {
        if (nums1[i] <= nums2[j]) {
            distance = Math.max(distance, j - i);
            j++;
        }
        else if (i < j) {
            i++;
        }
        else {
            i++;
            j++;
        }
    }

    return distance;
};

File: 1984-maximum-distance-between-a-pair-of-values/maximum-distance-between-a-pair-of-values.py
class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        ans = 0
        l1 = len(nums1)
        l2 = len(nums2)

        for i in range(l1):
            first = i
            last = l2 - 1

            while first <= last:
                mid = (first + last) // 2
                if nums2[mid] >= nums1[i]:
                    first = mid + 1
                else:
                    last = mid - 1

            ans = max(ans, last - i)
            
        return ans
